### PR TITLE
bpo-45600: Enhanced / clarified the docs for os.environ and os.environb

### PR DESCRIPTION
**Pull request by akulakov**
https://bugs.python.org/issue45600

Co-authored by Eryk Sun

### CODE REVIEW

File: Doc/library/os.rst [hunk 1]
```diff
@@ -170,9 +170,9 @@ process and user.
 
 .. data:: environ
 
-   A :term:`mapping` object representing the string environment. For example,
-   ``environ['HOME']`` is the pathname of your home directory (on some platforms),
-   and is equivalent to ``getenv("HOME")`` in C.
+   A :term:`mapping` object of key and value strings that represent the process
```
Comment (ambv) [inline Doc/library/os.rst]: This isn't valid English. You want to say "of string keys and values", or less tersely "where keys and values are strings" which I think reads better.

File: Doc/library/os.rst [hunk 2]
```diff
@@ -209,10 +209,10 @@ process and user.
 
 .. data:: environb
 
-   Bytes version of :data:`environ`: a :term:`mapping` object representing the
-   environment as byte strings. :data:`environ` and :data:`environb` are
-   synchronized (modify :data:`environb` updates :data:`environ`, and vice
-   versa).
+   Bytes version of :data:`environ`: a :term:`mapping` object of key and value
+   byte strings representing the process environment. :data:`environ` and
```
Comment (ambv) [inline Doc/library/os.rst]: Similarly, here you can say "where both keys and values are :class:`bytes` objects" (we avoid saying "byte strings" now, there is only a few cases of this left in the entire documentation).

Comment (akulakov) [reply Doc/library/os.rst]: I like "keys and values are strings", updated.

File: Doc/library/os.rst [hunk 3]
```diff
@@ -209,10 +210,10 @@ process and user.
 
 .. data:: environb
 
-   Bytes version of :data:`environ`: a :term:`mapping` object representing the
-   environment as byte strings. :data:`environ` and :data:`environb` are
-   synchronized (modify :data:`environb` updates :data:`environ`, and vice
-   versa).
+   Bytes version of :data:`environ`: a :term:`mapping` object where both keys
+   and values are :class:`bytes` objects representing the process environment.
+   :data:`environ` and :data:`environb` are synchronized (modify
```
Comment (ambv) [inline Doc/library/os.rst]:
```suggestion
   :data:`environ` and :data:`environb` are synchronized (modifying
```

This is Victor's 11 year old grammar mistake (all the way from 84ae1180063!). Let's kill two birds with one stone and fix this one too.

Comment (akulakov) [reply Doc/library/os.rst]: I've read this paragraph "carefully" yesterday but still missed it, hah!

**Comment by akulakov** [comment]
@ambv thanks for reviewing!

**Pull request merged**